src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1017,7 +1019,121 @@ else if (block instanceof BlockLiquid)
             return eyes < pos.getY() + 1 + filled;
         }
     }
+    
+    /*
+     * Predicate version of Entity inside material method
+     */
+    public static boolean isEntityInsideOfMaterial(Entity entity, Predicate<Material> predicate)
+    {
+        if (entity.getRidingEntity() instanceof EntityBoat)
+        {
+            return false;
+        }
+        else
+        {
+            BlockPos pos = new BlockPos(entity.posX, entity.posY + entity.getEyeHeight(), entity.posZ);
+            Material material = entity.world.getBlockState(pos).getMaterial();
+            if (predicate.test(material))
+            {
+                // DEBUG
+                if (entity instanceof EntityPlayer) System.out.println("Passed predicate test");
+                return isInsideOfMaterial(entity, pos);
+            }
+            else
+            {
+                return false;
+            }
+        }    
+    }
+    
+    public static Predicate<Material> isMaterialLiquid()
```

Comment:
I don't think these predicate definitions are needed. 
I think it's possible to easily define them like `Material::isLiquid` when you need one of these predicates.

Reply:
I only put them in as convenience, but yeah you can predicate directly. Okay I'll take em out.

Reply:
Took them out, and passed predicates directly, in latest commit.